src/internal/storage/fileset/fileset.go

```diff
@@ -1,42 +1,73 @@
 package fileset
 
 import (
+	"bytes"
 	"context"
+	"crypto/rand"
+	"database/sql/driver"
+	"encoding/hex"
 	"io"
 
 	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
 	"github.com/pachyderm/pachyderm/v2/src/internal/storage/chunk"
 	"github.com/pachyderm/pachyderm/v2/src/internal/storage/fileset/index"
-	"github.com/pachyderm/pachyderm/v2/src/internal/uuid"
 )
 
 // ID is the unique identifier for a fileset
-// TODO: change this to [16]byte, add ParseID, and HexString methods.
-type ID string
+type ID [16]byte
 
 func newID() ID {
-	return ID(uuid.NewWithoutDashes())
+	id := ID{}
+	if _, err := rand.Read(id[:]); err != nil {
+		panic(err)
+	}
+	return id
 }
 
 // ParseID parses a string into an ID or returns an error
 func ParseID(x string) (*ID, error) {
 	if len(x) < 32 {
 		return nil, errors.Errorf("string (%v) too short to be ID", x)
 	}
-	id := ID(x)
+	data, err := hex.DecodeString(x)
+	if err != nil {
+		return nil, err
+	}
+	id := ID{}
+	copy(id[:], data[:])
 	return &id, nil
 }
 
 // HexString returns the ID encoded with the hex alphabet.
 func (id ID) HexString() string {
-	return string(id)
+	return hex.EncodeToString(id[:])
 }
 
-// TrackerID returns the ID of the fileset's tracker object.
+// TrackerID returns the tracker ID for the fileset.
 func (id ID) TrackerID() string {
 	return TrackerPrefix + id.HexString()
 }
 
+// Scan implements sql.Scanner
+func (id *ID) Scan(src interface{}) error {
+	x, ok := src.([]byte)
+	if !ok {
+		return errors.Errorf("scanning fileset.ID: can't turn %T into fileset.ID", src)
+	}
+	x = bytes.Replace(x, []byte{'-'}, []byte{}, -1)
+	*id = ID{}
+	if len(x) < 32 {
```

Comment:
Maybe this should be a helper function that we can call to validate a hex encoded ID?

Reply:
what other place would we use that?

Reply:
Where I commented above. It seems like the validation should check for an exact length and be implemented in one place through a helper function.